pkg/kn/commands/broker/config_flags.go

```diff
@@ -0,0 +1,126 @@
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+type ConfigType int
+
+// Known config types for broker
+const (
+	ConfigMapType ConfigType = iota
+	SecretType
+	RabbitMqType
+)
+
+var (
+	// KReferenceMapping is mapping between the known config kinds to a basic
+	// default KReference value
+	KReferenceMapping = map[ConfigType]*duckv1.KReference{
+		ConfigMapType: {Kind: "ConfigMap", APIVersion: "v1"},
+		SecretType:    {Kind: "Secret", APIVersion: "v1"},
+		RabbitMqType:  {Kind: "RabbitmqCluster", APIVersion: "rabbitmq.com/v1beta1"},
+	}
+)
+
+// ConfigFlags represents the broker config
+type ConfigFlags struct {
+	BrokerConfig string
+}
+
+// Add is used to add the broker config flag to a command
+func (c *ConfigFlags) Add(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&c.BrokerConfig, "broker-config", "", "Reference to the configuration "+
+		"that specifies configuration options for this Broker. For example, a pointer to a ConfigMap, Secret, RabbitmqCluster etc."+
+		"The format for specifying the object is a colon separated string consisting of at most 3 substrings:\n"+
+		"kind:object-name:namespace=?,apiVersion=?,group=?\nThe third substring is optional and the following is also acceptable "+
+		"(in case of ConfigMap, Secret, and RabbitmqCluster kinds):\n"+
+		"kind:object-name")
+}
+
+// GetBrokerConfigReference parses the broker config
+// and return the appropriate KReference object
+func (c *ConfigFlags) GetBrokerConfigReference() (*duckv1.KReference, error) {
+	config := c.BrokerConfig
+	slices := strings.SplitN(config, ":", 3)
+	if len(slices) == 1 {
+		return &duckv1.KReference{
+			Kind:       "ConfigMap",
+			Name:       slices[0],
+			APIVersion: "v1",
+		}, nil
+	} else if len(slices) == 2 {
+		kind := slices[0]
+		name := slices[1]
+		kRef := getDefaultKReference(kind)
+		if kRef.APIVersion == "" {
+			return nil, fmt.Errorf("kind %q is unknown and APIVersion could not be determined", kind)
+		}
+		kRef.Name = name
+		return kRef, nil
+	} else {
+		kind := slices[0]
+		name := slices[1]
+		kRef := getDefaultKReference(kind)
+		kRef.Name = name
+
+		params := strings.Split(slices[2], ",")
+		if len(params) == 1 && !strings.Contains(params[0], "=") {
+			kRef.Namespace = params[0]
+			return kRef, nil
+		}
+		mappedOptions, err := util.MapFromArray(params, "=")
+		if err != nil {
+			return nil, err
+		}
+		for k, v := range mappedOptions {
+			switch strings.ToLower(k) {
+			case "namespace":
+				kRef.Namespace = v
+			case "group":
+				kRef.Group = v
+			case "apiversion":
+				kRef.APIVersion = v
+			default:
+				return nil, fmt.Errorf("incorrect field %q. Please specify any of the following: Namespace, Group, APIVersion", k)
+			}
+		}
+		if kRef.APIVersion == "" {
+			return nil, fmt.Errorf("kind %q is unknown and APIVersion could not be determined", kind)
+		}
+		return kRef, nil
+	}
+}
+
+func getDefaultKReference(kind string) *duckv1.KReference {
+	k := strings.ToLower(kind)
+	switch k {
+	case "cm", "configmap":
```

Comment:
could we make that switch more dynamic ? I.e. already add the list of shortcut prefixes to `KReferenceMapping` and then either iterate of the list to find the entry or build up a map before with the shortcut prefix as key, pointing to the KReferenceMapping entry (i.e. I would create an own Struct for the mapping with a KReference and a string array as member for KRerferenceMapping.

The advantage would be that all the relevant parts (reference + prefixes) are defined in a single place (i.e. where KReferenceMapping is defined)

Reply:
Yes, sounds good. I'll do that.